### PR TITLE
Refactor output writer

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

* Refactor output writer and combine similar methods and operations.
* Refactor output-related tests, combining similar tests

* adding test cover on outputwriter package
* fix bugs discovered in content

* combine handling comments and their content

### CODE REVIEW

File: utils/outputwriter/outputcontent_test.go [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+)
+
+func TestIsFrogbotSummaryComment(t *testing.T) {
+	testCases := []struct {
+		name    string
+		comment string
+		cases   []OutputTestCase
+	}{
+		{
+			name:    "Not Summary comment",
```
Comment (yahavi) [inline utils/outputwriter/outputcontent_test.go:21]:
```suggestion
			name:    "No Summary comment",
```

File: utils/outputwriter/simplifiedoutput_test.go [hunk 2]
```diff
@@ -1,449 +1,233 @@
...
-	}
-
-	testCases := []testCase{
+func TestSimpleOutputFlags(t *testing.T) {
+	testCases := []struct {
+		name         string
+		entitled     bool
+		showCaColumn bool
+	}{
+		{
```
Comment (yahavi) [inline utils/outputwriter/simplifiedoutput_test.go:15]: When the test cases are short and simple, I think it is better to inline them:
```go
		{name: "entitled", entitled: true, showCaColumn: false},
		{name: "not entitled", entitled: false, showCaColumn: false},
		{name: "entitled with ca column", entitled: true, showCaColumn: true},
		{name: "not entitled with ca column", entitled: false, showCaColumn: true},
```

File: utils/outputwriter/standardoutput.go [hunk 3]
```diff
@@ -5,7 +5,6 @@ import (
 	"strings"
 
 	"github.com/jfrog/froggit-go/vcsutils"
-	"github.com/jfrog/jfrog-cli-core/v2/xray/formats"
 )
 
 type StandardOutput struct {
```
Comment (yahavi) [inline utils/outputwriter/standardoutput.go:10]: The "StandardOutput" and the "SimplifiedOutput" can inherit from the same struct. That will help to remove redundancy in variables and methods.

Comment (yahavi) [inline utils/outputwriter/standardoutput.go:10]: StandardOutput is not the best name IMO.
I suggest renaming it to "DecoratedOutput".

File: utils/outputwriter/markdowntable.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+	Name         string
+	BuildType    MarkdownColumnType
+	DefaultValue string
+}
+
+type CellData []string
+
+func NewCellData(values ...string) CellData {
+	if len(values) == 0 {
+		return CellData{""}
```
[the remark this answers is not in the document]
Comment (attiasas) [reply utils/outputwriter/markdowntable.go:45]: @yahavi Yes, I added a comment explaining

File: utils/outputwriter/standardoutput.go (hunk 3 above)
Comment (attiasas) [reply utils/outputwriter/standardoutput.go:8]: @yahavi, I believe "StandardOutput" is a more fitting term. This is a Markdown interface, and "Standard" implies that it adheres to all the available features in Markdown, essentially representing the expected output. So, in my view, it's not just a decoration but rather the standard output in Markdown.

**Pull request merged**